### ISSUE TITLE
W1b: `RandomBoxBlur` rejects the `BorderType`, int and uppercase `border_type` spellings its sibling blurs accept

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4590]
### Describe the bug

`RandomBoxBlur` is the only one of the three padding blurs that does not normalize `border_type` through
`BorderType.get()`. It stores the raw argument (`kornia/augmentation/_2d/intensity/box_blur.py:91`):

```python
self.flags = {"kernel_size": kernel_size, "border_type": border_type, "normalized": normalized}
```

while `RandomGaussianBlur` (`gaussian_blur.py:110`) and `RandomMotionBlur` (`motion_blur.py:120`) both do
`"border_type": BorderType.get(border_type)`.

The result is that three spellings its siblings accept fail on `RandomBoxBlur`, each with a different and
unhelpful error:

| `border_type` | `RandomBoxBlur` | `RandomGaussianBlur` | `RandomMotionBlur` |
|---|---|---|---|
| `"reflect"` | ok | ok | ok |
| `BorderType.REFLECT` | `BaseError` | ok | ok |
| `1` | `BaseError` | ok | ok |
| `"REFLECT"` | `NotImplementedError` | ok | ok |
| `"bogus"` | `BaseError` | `KeyError` | `KeyError` |

This is easy to hit because `RandomMotionBlur`'s own `Args` advertises the integer form
("CONSTANT = 0, REFLECT = 1, REPLICATE = 2, CIRCULAR = 3"), and the three classes are otherwise documented
as a matched set — the batch 6c convention blocks for box and Gaussian blur are word for word identical.

### Reproduction steps

```python
import torch
from kornia.augmentation import RandomBoxBlur, RandomGaussianBlur
from kornia.constants import BorderType

img = torch.rand(1, 1, 9, 9)
RandomGaussianBlur((3, 3), (1.0, 1.0), border_type=BorderType.REFLECT, p=1.0)(img)   # ok
RandomBoxBlur((3, 3), border_type=BorderType.REFLECT, p=1.0)(img)                    # BaseError
RandomBoxBlur((3, 3), border_type=1, p=1.0)(img)                                     # BaseError
RandomBoxBlur((3, 3), border_type="REFLECT", p=1.0)(img)                             # NotImplementedError
```

### Expected behavior

`RandomBoxBlur` accepts the same `border_type` spellings as `RandomGaussianBlur` and `RandomMotionBlur`.

The fix is one line — normalize in `__init__` and pass `.name.lower()` to `box_blur`, exactly as
`RandomGaussianBlur.apply_transform` already does — plus widening the annotation from `str` to
`Union[int, str, BorderType]` to match `RandomMotionBlur`.

### Environment

kornia at `8aca2838f` (PR #4566 head), torch 2.14.0, Python 3.11.14, macOS arm64, CPU.
Found while reviewing #4566. No documentation change is proposed there: per kornia policy this is a wart
to fix rather than to write down.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
